05.ls/lib/ls_practice.rb

```diff
@@ -1,43 +1,139 @@
-      opt.parse!(ARGV)
+  def file_info_hash(stat_files, files)
+    # 1度のループで最大長を取得する
+    max_length_hash = { hard_link: 0, user_name: 0, group_name: 0, size: 0 }
+    file_info_hash_array = stat_files.map.with_index do |stat_file, index|
+      hard_link = stat_file.nlink.to_s
+      max_length_hash[:hard_link] = hard_link.length if hard_link.length > max_length_hash[:hard_link]
+      user_name = Etc.getpwuid(stat_file.uid).name
+      max_length_hash[:user_name] = user_name.length if user_name.length > max_length_hash[:user_name]
+      group_name = Etc.getgrgid(stat_file.gid).name
+      max_length_hash[:group_name] = group_name.length if group_name.length > max_length_hash[:group_name]
+      size = stat_file.size.to_s
+      max_length_hash[:size] = size.length if size.length > max_length_hash[:size]
+      { type: file_type(stat_file),
+        permission: format_permission(stat_file),
+        hard_link: hard_link,
+        user_name: user_name,
+        group_name: group_name,
+        size: size,
+        last_modified_time: stat_file.mtime.strftime('%b %d %H:%M'),
+        name: name_with_symlink(files[index]) }
     end
-    options
+    { max_length_hash: max_length_hash, file_info_hash_array: file_info_hash_array }
   end
 
-  def create_files_and_folders_text
-    files = files_and_folders
-    files = files.reverse if reverse_order?
+  def file_info_array(file_info_hash_array, max_length_hash)
+    file_info_hash_array.map do |file_info|
+      format_file_info(file_info, max_length_hash)
+    end
+  end
 
-    column_item_number = files.size / MAX_COLUMN_NUMBER
+  def format_file_info(file_info_hash, max_length_hash)
+    single_white_spaces = ' '
+    two_white_spaces = '  '
+    text = file_info_hash[:type] + file_info_hash[:permission]
+    text += two_white_spaces
+    text += file_info_hash[:hard_link].rjust(max_length_hash[:hard_link])
+    text += single_white_spaces
+    text += file_info_hash[:user_name].ljust(max_length_hash[:user_name])
+    text += two_white_spaces
+    text += file_info_hash[:group_name].ljust(max_length_hash[:group_name])
+    text += two_white_spaces
+    text += file_info_hash[:size].rjust(max_length_hash[:size])
+    text += single_white_spaces
+    text += file_info_hash[:last_modified_time]
+    text += single_white_spaces
+    text += file_info_hash[:name]
+    text
+  end
 
-    max_text_length = files.map(&:length).max + WHITE_SPACE_INDENT_LENGTH
+  def convert_to_stat_files(files)
+    files.map do |file|
+      if FileTest.symlink?(file)
+        File.lstat(file)
+      else
+        File.stat(file)
+      end
+    end
+  end
 
-    text_array = []
+  def name_with_symlink(file)
+    if FileTest.symlink?(file)
+      "#{file} -> #{File.readlink(file)}"
+    else
+      file.to_s
+    end
+  end
+
+  def total_blocks(stat_files)
+    stat_files.sum(&:blocks)
+  end
+
+  def file_type(file)
+    if file.ftype == 'file'
+      '-'
+    else
+      file.ftype[0]
+    end
```

Comment:
このままでも問題はないですが、`if`と`else`の中身が両方一行の場合は三項演算子でかけますね。参考までに。
```suggestion
    file.ftype == 'file' ? '-' : file.ftype[0]
```